### PR TITLE
docker-credential-gcr: init at 1.4.3

### PR DESCRIPTION
**Pull request by suvash**
Add a (Golang based) binary that provides authentication to Docker repositories hosted on Google Cloud Registry (GCR). This tool is required when using https://gcr.io based Docker repositories hosted on GCP, making use of various Google Oauth features to authenticate.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by suvash** [comment]
Only installed NixOS over the weekend, and planning to get back to using it. Obviously, I've forgotten a lot of how it works. Let me know if there's anything missing/wrong with this package request.

Also, is it possible to add the local nixpkgs repo as some form of source (like a channel) and install this tool declaratively via `configuration.nix`, for eg. I already have channels setup for `unstable`, so I can pick out packages from that channels with a prefix for eg. `unstable.spotify`, can I do the same for local repos with some sort of prefix until this lands in `unstable` ? I'm sure this is possible, just not sure how to best do it as a beginner. Any pointers would be much appreciated.
Something like either of this would be super.
- https://nixos.wiki/wiki/Cheatsheet#Upgrading_individual_packages_to_a_different_channel
- https://nixos.wiki/wiki/FAQ#How_can_I_install_a_package_from_unstable_while_remaining_on_the_stable_channel.3F

File: pkgs/tools/admin/docker-credential-gcr/default.nix [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  goPackagePath = "github.com/GoogleCloudPlatform/docker-credential-gcr";
+
+  src = fetchFromGitHub {
+    owner = "GoogleCloudPlatform";
+    repo = "docker-credential-gcr";
+    rev = "v${version}";
+    sha256 = "1xb88xjyyrdmjcgfv7fqdkv1ip3dpzsdif5vm7vkqvn83s5wj5df";
+  };
+
+  goDeps = ./deps.nix;
```
Comment (nlewo) [inline pkgs/tools/admin/docker-credential-gcr/default.nix]: You could remove it since it is empty

**Review by nlewo — CHANGES_REQUESTED** [review]

**Comment by suvash** [comment]
@nlewo Updated !

**Comment by nlewo** [comment]
@GrahamcOfBorg build docker-credential-gcr

**Comment by nlewo** [comment]
Thanks!

**Pull request merged**